Warnings regression in coreclr build

These warnings are newly seen in the coreclr product build (specifically, System.Private.Corelib.dll build):

```
BUILD: Commencing build of System.Private.CoreLib for Windows_NT.x64.Checked
Tools are already initialized.
Running: dotnet restore /nologo /verbosity:minimal /clp:Summary /nodeReuse:false          /l:BinClashLogger,Tools/Microsoft.DotNet.Build.Tasks.dll;LogFile=binclash.log          /p:RestoreDefaultOptimizationDataPackage=false /p:PortableBuild=true          /p:UsePartialNGENOptimization=false /maxcpucount /p:IncludeRestoreOnlyProjects=true /p:ArcadeBuild=true          F:\workspace\_work\1\s\src\build.proj          /flp:Verbosity=normal;LogFile="F:\workspace\_work\1\s\bin\Logs\System.Private.CoreLib_Windows_NT__x64__Checked.log" /flp1:WarningsOnly;LogFile="F:\workspace\_work\1\s\bin\Logs\System.Private.CoreLib_Windows_NT__x64__Checked.wrn" /flp2:ErrorsOnly;LogFile="F:\workspace\_work\1\s\bin\Logs\System.Private.CoreLib_Windows_NT__x64__Checked.err" /p:__BuildOS=Windows_NT /p:__BuildType=Checked /p:__BuildArch=x64 /p:RestoreDuringBuild=true   /p:BuildManagedTools=true /p:BuildNugetPackage=true
Restoring packages for C:\Users\vsagent\AppData\Local\Temp\06643f3e-3b83-46b3-9bbf-78d881a2cfc5...
Installing Microsoft.Build.Traversal 1.0.52.
Restoring packages for C:\Users\vsagent\AppData\Local\Temp\06643f3e-3b83-46b3-9bbf-78d881a2cfc5...
Installing Microsoft.DotNet.Arcade.Sdk 1.0.0-beta.19229.8.
F:\workspace\_work\1\s\dir.props(5,3): warning MSB4011: "F:\workspace\_work\1\s\dir.common.props" cannot be imported again. It was already imported at "F:\workspace\_work\1\s\Directory.Build.props (4,3)". This is most likely a build authoring error. This subsequent import will be ignored. [F:\workspace\_work\1\s\src\build.proj]
F:\workspace\_work\1\s\dependencies.props(14,3): warning MSB4011: "F:\workspace\_work\1\s\eng\Versions.props" cannot be imported again. It was already imported at "C:\Users\vsagent\.nuget\packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19229.8\tools\DefaultVersions.props (17,3)". This is most likely a build authoring error. This subsequent import will be ignored. [F:\workspace\_work\1\s\src\build.proj]
F:\workspace\_work\1\s\dir.props(5,3): warning MSB4011: "F:\workspace\_work\1\s\dir.common.props" cannot be imported again. It was already imported at "F:\workspace\_work\1\s\Directory.Build.props (4,3)". This is most likely a build authoring error. This subsequent import will be ignored. [F:\workspace\_work\1\s\src\build.proj]
F:\workspace\_work\1\s\dependencies.props(14,3): warning MSB4011: "F:\workspace\_work\1\s\eng\Versions.props" cannot be imported again. It was already imported at "C:\Users\vsagent\.nuget\packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19229.8\tools\DefaultVersions.props (17,3)". This is most likely a build authoring error. This subsequent import will be ignored. [F:\workspace\_work\1\s\src\build.proj]
F:\workspace\_work\1\s\Directory.Build.props(4,3): warning MSB4011: "F:\workspace\_work\1\s\dir.common.props" cannot be imported again. It was already imported at "F:\workspace\_work\1\s\dir.props (5,3)". This is most likely a build authoring error. This subsequent import will be ignored. [F:\workspace\_work\1\s\src\.nuget\init\init.csproj]
C:\Users\vsagent\.nuget\packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19229.8\tools\DefaultVersions.props(17,3): warning MSB4011: "F:\workspace\_work\1\s\eng\Versions.props" cannot be imported again. It was already imported at "F:\workspace\_work\1\s\dependencies.props (14,3)". This is most likely a build authoring error. This subsequent import will be ignored. [F:\workspace\_work\1\s\src\.nuget\init\init.csproj]
F:\workspace\_work\1\s\dir.props(5,3): warning MSB4011: "F:\workspace\_work\1\s\dir.common.props" cannot be imported again. It was already imported at "F:\workspace\_work\1\s\Directory.Build.props (4,3)". This is most likely a build authoring error. This subsequent import will be ignored. [F:\workspace\_work\1\s\src\ToolBox\SOS\NETCore\SOS.NETCore.csproj]
F:\workspace\_work\1\s\dir.props(5,3): warning MSB4011: "F:\workspace\_work\1\s\dir.common.props" cannot be imported again. It was already imported at "F:\workspace\_work\1\s\Directory.Build.props (4,3)". This is most likely a build authoring error. This subsequent import will be ignored. [F:\workspace\_work\1\s\src\tools\r2rdump\R2RDump.csproj]
F:\workspace\_work\1\s\dir.props(5,3): warning MSB4011: "F:\workspace\_work\1\s\dir.common.props" cannot be imported again. It was already imported at "F:\workspace\_work\1\s\Directory.Build.props (4,3)". This is most likely a build authoring error. This subsequent import will be ignored. [F:\workspace\_work\1\s\src\System.Private.CoreLib\System.Private.CoreLib.csproj]
F:\workspace\_work\1\s\dependencies.props(14,3): warning MSB4011: "F:\workspace\_work\1\s\eng\Versions.props" cannot be imported again. It was already imported at "C:\Users\vsagent\.nuget\packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19229.8\tools\DefaultVersions.props (17,3)". This is most likely a build authoring error. This subsequent import will be ignored. [F:\workspace\_work\1\s\src\ToolBox\SOS\NETCore\SOS.NETCore.csproj]
F:\workspace\_work\1\s\dependencies.props(14,3): warning MSB4011: "F:\workspace\_work\1\s\eng\Versions.props" cannot be imported again. It was already imported at "C:\Users\vsagent\.nuget\packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19229.8\tools\DefaultVersions.props (17,3)". This is most likely a build authoring error. This subsequent import will be ignored. [F:\workspace\_work\1\s\src\tools\r2rdump\R2RDump.csproj]
F:\workspace\_work\1\s\dependencies.props(14,3): warning MSB4011: "F:\workspace\_work\1\s\eng\Versions.props" cannot be imported again. It was already imported at "C:\Users\vsagent\.nuget\packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19229.8\tools\DefaultVersions.props (17,3)". This is most likely a build authoring error. This subsequent import will be ignored. [F:\workspace\_work\1\s\src\System.Private.CoreLib\System.Private.CoreLib.csproj]
```

@AaronRobinsonMSFT This is what I see from a build this morning https://dev.azure.com/dnceng/public/_build/results?buildId=225141.  The warnings look similar, so I assumed this still needs work.

```
BUILD: Commencing build of System.Private.CoreLib for Windows_NT.x64.Checked
Installing dotnet using Arcade...
running: powershell -NoProfile -ExecutionPolicy unrestricted -Command ". F:\workspace\_work\1\s\eng\configure-toolset.ps1; . F:\workspace\_work\1\s\eng\common\tools.ps1; InitializeBuildTool"

Name                           Value                                                                                   
----                           -----                                                                                   
Path                           F:\workspace\_work\1\s\.dotnet\dotnet.exe                                               
Command                        msbuild                                                                                 
Tool                           dotnet                                                                                  
Framework                      netcoreapp2.1                                                                           


Running: dotnet restore /nologo /verbosity:minimal /clp:Summary /nodeReuse:false          /p:PortableBuild=true /maxcpucount /p:IncludeRestoreOnlyProjects=true /p:ArcadeBuild=true          F:\workspace\_work\1\s\src\build.proj          /flp:Verbosity=normal;LogFile="F:\workspace\_work\1\s\bin\Logs\System.Private.CoreLib_Windows_NT__x64__Checked.log" /flp1:WarningsOnly;LogFile="F:\workspace\_work\1\s\bin\Logs\System.Private.CoreLib_Windows_NT__x64__Checked.wrn" /flp2:ErrorsOnly;LogFile="F:\workspace\_work\1\s\bin\Logs\System.Private.CoreLib_Windows_NT__x64__Checked.err" /p:__BuildOS=Windows_NT /p:__BuildType=Checked /p:__BuildArch=x64 /p:RestoreDuringBuild=true   /p:BuildManagedTools=true     /p:ContinuousIntegrationBuild=true
F:\workspace\_work\1\s\dir.props(3,3): warning MSB4011: "F:\workspace\_work\1\s\dir.common.props" cannot be imported again. It was already imported at "F:\workspace\_work\1\s\Directory.Build.props (4,3)". This is most likely a build authoring error. This subsequent import will be ignored. [F:\workspace\_work\1\s\src\build.proj]
F:\workspace\_work\1\s\dependencies.props(14,3): warning MSB4011: "F:\workspace\_work\1\s\eng\Versions.props" cannot be imported again. It was already imported at "F:\workspace\_work\1\s\.packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19312.21\tools\DefaultVersions.props (17,3)". This is most likely a build authoring error. This subsequent import will be ignored. [F:\workspace\_work\1\s\src\build.proj]
F:\workspace\_work\1\s\dir.props(3,3): warning MSB4011: "F:\workspace\_work\1\s\dir.common.props" cannot be imported again. It was already imported at "F:\workspace\_work\1\s\Directory.Build.props (4,3)". This is most likely a build authoring error. This subsequent import will be ignored. [F:\workspace\_work\1\s\src\build.proj]
F:\workspace\_work\1\s\dependencies.props(14,3): warning MSB4011: "F:\workspace\_work\1\s\eng\Versions.props" cannot be imported again. It was already imported at "F:\workspace\_work\1\s\.packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19312.21\tools\DefaultVersions.props (17,3)". This is most likely a build authoring error. This subsequent import will be ignored. [F:\workspace\_work\1\s\src\build.proj]
F:\workspace\_work\1\s\dir.props(3,3): warning MSB4011: "F:\workspace\_work\1\s\dir.common.props" cannot be imported again. It was already imported at "F:\workspace\_work\1\s\Directory.Build.props (4,3)". This is most likely a build authoring error. This subsequent import will be ignored. [F:\workspace\_work\1\s\src\ToolBox\SOS\NETCore\SOS.NETCore.csproj]
F:\workspace\_work\1\s\dependencies.props(14,3): warning MSB4011: "F:\workspace\_work\1\s\eng\Versions.props" cannot be imported again. It was already imported at "F:\workspace\_work\1\s\.packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19312.21\tools\DefaultVersions.props (17,3)". This is most likely a build authoring error. This subsequent import will be ignored. [F:\workspace\_work\1\s\src\ToolBox\SOS\NETCore\SOS.NETCore.csproj]
F:\workspace\_work\1\s\dir.props(3,3): warning MSB4011: "F:\workspace\_work\1\s\dir.common.props" cannot be imported again. It was already imported at "F:\workspace\_work\1\s\Directory.Build.props (4,3)". This is most likely a build authoring error. This subsequent import will be ignored. [F:\workspace\_work\1\s\src\tools\runincontext\runincontext.csproj]
F:\workspace\_work\1\s\dependencies.props(14,3): warning MSB4011: "F:\workspace\_work\1\s\eng\Versions.props" cannot be imported again. It was already imported at "F:\workspace\_work\1\s\.packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19312.21\tools\DefaultVersions.props (17,3)". This is most likely a build authoring error. This subsequent import will be ignored. [F:\workspace\_work\1\s\src\tools\runincontext\runincontext.csproj]
F:\workspace\_work\1\s\dir.props(3,3): warning MSB4011: "F:\workspace\_work\1\s\dir.common.props" cannot be imported again. It was already imported at "F:\workspace\_work\1\s\Directory.Build.props (4,3)". This is most likely a build authoring error. This subsequent import will be ignored. [F:\workspace\_work\1\s\src\tools\r2rdump\R2RDump.csproj]
F:\workspace\_work\1\s\dependencies.props(14,3): warning MSB4011: "F:\workspace\_work\1\s\eng\Versions.props" cannot be imported again. It was already imported at "F:\workspace\_work\1\s\.packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19312.21\tools\DefaultVersions.props (17,3)". This is most likely a build authoring error. This subsequent import will be ignored. [F:\workspace\_work\1\s\src\tools\r2rdump\R2RDump.csproj]
F:\workspace\_work\1\s\dir.props(3,3): warning MSB4011: "F:\workspace\_work\1\s\dir.common.props" cannot be imported again. It was already imported at "F:\workspace\_work\1\s\Directory.Build.props (4,3)". This is most likely a build authoring error. This subsequent import will be ignored. [F:\workspace\_work\1\s\src\System.Private.CoreLib\System.Private.CoreLib.csproj]
F:\workspace\_work\1\s\dependencies.props(14,3): warning MSB4011: "F:\workspace\_work\1\s\eng\Versions.props" cannot be imported again. It was already imported at "F:\workspace\_work\1\s\.packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19312.21\tools\DefaultVersions.props (17,3)". This is most likely a build authoring error. This subsequent import will be ignored. [F:\workspace\_work\1\s\src\System.Private.CoreLib\System.Private.CoreLib.csproj]
  Restore completed in 62.09 ms for F:\workspace\_work\1\s\src\build.proj.
  Restore completed in 566.04 ms for F:\workspace\_work\1\s\src\System.Private.CoreLib\System.Private.CoreLib.csproj.
  Restore completed in 446.44 ms for F:\workspace\_work\1\s\src\tools\runincontext\runincontext.csproj.
F:\workspace\_work\1\s\src\ToolBox\SOS\NETCore\SOS.NETCore.csproj : warning NU1604: Project dependency System.Reflection.Metadata does not contain an inclusive lower bound. Include a lower bound in the dependency version to ensure consistent restore results. [F:\workspace\_work\1\s\src\build.proj]
F:\workspace\_work\1\s\src\ToolBox\SOS\NETCore\SOS.NETCore.csproj : warning NU1701: Package 'System.Collections.Immutable 1.1.36' was restored using '.NETFramework,Version=v4.6.1' instead of the project target framework '.NETCoreApp,Version=v2.0'. This package may not be fully compatible with your project. [F:\workspace\_work\1\s\src\build.proj]
F:\workspace\_work\1\s\src\ToolBox\SOS\NETCore\SOS.NETCore.csproj : warning NU1701: Package 'System.Reflection.Metadata 1.0.21' was restored using '.NETFramework,Version=v4.6.1' instead of the project target framework '.NETCoreApp,Version=v2.0'. This package may not be fully compatible with your project. [F:\workspace\_work\1\s\src\build.proj]
  Restore completed in 2.01 sec for F:\workspace\_work\1\s\src\ToolBox\SOS\NETCore\SOS.NETCore.csproj.
  Restore completed in 3.31 sec for F:\workspace\_work\1\s\src\tools\r2rdump\R2RDump.csproj.
```

--- a/dependencies.props
+++ b/dependencies.props
@@ -11,7 +11,8 @@
     </RestoreSources>
   </PropertyGroup>
 
-  <Import Project="eng/Versions.props" />
+  <!-- [ARCADE REMOVE] Versions.props is imported by Arcade SDK targets when building with Arcade -->
+  <Import Project="eng/Versions.props" Condition="'$(ArcadeBuild)' != 'true'" />
 
   <!-- Source of truth for dependency tooling: the commit hash of the dotnet/versions master branch as of the last auto-upgrade. -->
   <PropertyGroup>

--- a/dir.props
+++ b/dir.props
@@ -1,6 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project>
-  <Import Project="dir.common.props" />
+  <!-- dir.common.props is imported by Directory.Build.props when building with Arcade -->
+  <Import Project="dir.common.props" Condition="'$(ArcadeBuild)' != 'true'" />
 
   <!-- [ARCADE REMOVE] This entire file can be removed. Properties set by this file should be moved into
        dir.common.props and Directory.Build.props as appropriate and projects should stop importing this. -->

